tests/tensorflow/test_printing.py

```diff
@@ -79,3 +79,47 @@ def test_unknown_shape():
     expected_out = 'Tensor(Placeholder):0,\tshape=Unknown\t"A:0"\n'
 
     assert std_out.getvalue() == expected_out.lstrip()
+
+
+@pytest.mark.usefixtures("run_with_tensorflow")
+@run_in_graph_mode
+def test_shape_index():
+    """Make sure graph indexing functions as expected."""
+
+    A = tf.compat.v1.placeholder('float', name='A',
+                                 shape=tf.TensorShape([None, None]))
+    x = tf.compat.v1.placeholder('float', name='x',
+                                 shape=tf.TensorShape([None, 1]))
+    y = tf.multiply(1.0, x, name='y')
+
+    z = tf.matmul(A, tf.add(y, y, name='x_p_y'), name='A_dot')
+
+    range_out = io.StringIO()
+    with redirect_stdout(range_out):        
+        tf_dprint(z, depth_lower=1, depth_upper=4)
```

Comment:
Try `depth_upper >= 5`; we want a test that has upper-bound trucation across more than one term that has 
a non-trivial subgraph (i.e. more than one "..." at the same depth).

Reply:
Updated to use `depth_upper = 5`.